src/avram/migrator/create_table_statement.cr

```diff
@@ -78,9 +78,15 @@ class Avram::Migrator::CreateTableStatement
     {% if type_declaration.type.is_a?(Union) %}
       {% type = type_declaration.type.types.first %}
       {% nilable = true %}
+      {% array = false %}
+    {% elsif type_declaration.type.is_a?(Generic) %}
+      {% type = type_declaration.type.type_vars.first %}
+      {% nilable = false %}
+      {% array = true %}
     {% else %}
       {% type = type_declaration.type %}
       {% nilable = false %}
+      {% array = false %}
     {% end %}
 
     rows << Avram::Migrator::Columns::{{ type }}Column.new(
```

Comment:
I had an idea to clean up the defaults. Right now we have to remember to add `Array(TheType)` to each column. It also incorrectly would allow array defaults for non-array types and vice versa. 

Instead we could make the Column types generic and pass the default type to it. An example will be clear (I hope :P)

```crystal
# What we have now
rows << Avram::Migrator::Columns::{{ type }}Column.new(

# Proposed change 
rows << Avram::Migrator::Columns::{{ type }}Column({% if array %}Array({% end %}{{ allowed_default_type }}{% if array %}){% end %}).new(
```

Then in the column classes:

```crystal
module Avram::Migrator::Columns
   class BoolColumn(T) < Base
     @default : T | Nil = nil
```

So now the default type matches the array type explicitly, and also removes some duplication. Does that make sense?

Reply:
Yeah that makes sense. Is that a change you want me to include in this PR?

Reply:
Yeah I think that'd be good to have so we make sure the types for defaults line up correctly